### PR TITLE
feat: Profile setup (#40012)

### PR DESCRIPTION
**Pull request by nx-mukesh**

### CODE REVIEW

File: src/controller/user/Controller.ts [hunk 1]
```diff
@@ -87,13 +129,11 @@ class User {
...
-          data: userData,
-          deleted_User: toDeleteUser,
-        });
+      return res.status(200).send({
+        message: 'user deleted successfully',
+        data: userData,
+        deleted_User: toDeleteUser,
+      });
     } catch (error) {
       return res
```
Comment (namanpar) [inline src/controller/user/Controller.ts]: one try-catch is enough for function and throwing error. You need userRepostiory method for CRUD operation not mock one.

**Review by namanpar — CHANGES_REQUESTED** [review]

**Comment by namanpar** [comment]
Raise the pr according to task. Review your own pr first

File: src/controller/user/Controller.ts (hunk 1 above)
Comment (nx-mukesh) [reply src/controller/user/Controller.ts]: fixed all bugs now working find.

File: src/controller/user/Controller.ts [hunk 2]
```diff
@@ -1,47 +1,70 @@
...
 
-class User {
-  //   get User Details
-  get(req: Request, res: Response, next: NextFunction) {
-    const userData = [
-      { id: '1', name: 'John Milton', role: 'Author', address: 'Washington' },
-      { id: '2', name: 'Thomas Crew', role: 'Editor', address: 'New York' },
-      { id: '3', name: 'Selena peter', role: 'Singer', address: 'Alaska' },
-    ];
+const userRepository: UserRepository = new UserRepository();
```
Comment (namanpar) [inline src/controller/user/Controller.ts:7]: instantiate it inside class

File: src/controller/user/Controller.ts [hunk 3]
```diff
@@ -87,19 +104,30 @@ class User {
...
+        status: 200,
+      });
     } catch (error) {
-      return res
-        .status(500)
-        .send({ err: 'server error', message: 'Something went Wrong' });
+      return res.status(500).send({ err: 'Server Error', message: 'Something went wrong' });
     }
   }
 }
```
Comment (namanpar) [inline src/controller/user/Controller.ts:122]: On try-catch is enough. If data is missing or condition fails then throw error with proper message. Use catch block to catch the error and return. No need return error using multiple time using res.send()

**Review by namanpar — CHANGES_REQUESTED** [review]
Read and do the changes in pr according to the comment mentioned

**Review by namanpar — APPROVED** [review]

**Pull request merged**